sbor/src/codec/result.rs

```diff
@@ -51,3 +51,25 @@ impl<X: CustomTypeId, D: Decoder<X>, T: Decode<X, D>, E: Decode<X, D>> Decode<X,
         }
     }
 }
+
+#[cfg(feature = "schema")]
+impl<C: CustomTypeKind<GlobalTypeId>, T: Describe<C>, E: Describe<C>> Describe<C> for Result<T, E> {
+    const TYPE_ID: GlobalTypeId = GlobalTypeId::novel("Result", &[T::TYPE_ID, E::TYPE_ID]);
+
+    fn type_data() -> Option<TypeData<C, GlobalTypeId>> {
+        #[allow(unused_imports)]
+        use crate::rust::borrow::ToOwned;
+        Some(TypeData::named_enum(
+            "Result",
+            crate::rust::collections::btree_map::btreemap![
```

Comment:
We should probably use array to retain the original order.

Reply:
I originally used `vec` here... but then decided that at a typekind level, we don't actually care about order, and canonically forcing an ordering by variant key feels cleaner.

And I'm perfectly comfortable with losing original order at type reconstruction time. Type reconstruction for me is about creating DTOs rather than getting a perfect one-for-one mapping of code.

Are you happy with this call?

Reply:
It's going to be tricky if we use `u8` as descriminator,  so we don't know how to encode the generated enum types.

Reply:
Discussed on call - we can keep using btreemap when we pivot to `u8`, as this allows for us to eg have a variant with discriminants 5 and 7, and not necessarily start from 0.